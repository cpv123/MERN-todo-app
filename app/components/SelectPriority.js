import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Select from '@material-ui/core/Select';
import MenuItem from '@material-ui/core/MenuItem';
import Input from '@material-ui/core/Input';
import InputLabel from '@material-ui/core/InputLabel';

export default class SelectPriority extends Component {
    render() {
        return (
            <div>
                <InputLabel style={{ 'fontSize': 12 }}>
                    Priority
                </InputLabel>
                <Select
                    value={this.props.value}
                    onChange={this.props.onChangePriority}
                    >
                    <MenuItem value={"high"}>High</MenuItem>
                    <MenuItem value={"medium"}>Medium</MenuItem>
                    <MenuItem value={"low"}>Low</MenuItem>
                </Select>
            </div>
        )
    }
}