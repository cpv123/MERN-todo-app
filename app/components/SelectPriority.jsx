import React from 'react';
import PropTypes from 'prop-types';
import Select from '@material-ui/core/Select';
import MenuItem from '@material-ui/core/MenuItem';
import InputLabel from '@material-ui/core/InputLabel';

const options = ['High', 'Medium', 'Low'];

const SelectPriority = ({ value, onChangePriority }) => (
  <div style={{ display: 'flex', flexDirection: 'column' }}>
    <InputLabel style={{ fontSize: 12 }}>
      Priority
    </InputLabel>
    <Select
      value={value}
      onChange={onChangePriority}
      autoWidth={false}
    >
      {
          options.map(option => (
            <MenuItem value={option} key={option}>
              {option}
            </MenuItem>
          ))
      }
    </Select>
  </div>
);

SelectPriority.propTypes = {
  value: PropTypes.string.isRequired,
  onChangePriority: PropTypes.func.isRequired,
};

export default SelectPriority;
