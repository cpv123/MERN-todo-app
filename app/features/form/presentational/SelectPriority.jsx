import React from 'react';
import PropTypes from 'prop-types';
import Select from '@material-ui/core/Select';
import MenuItem from '@material-ui/core/MenuItem';
import InputLabel from '@material-ui/core/InputLabel';

export default function SelectPriority(props) {
  const { value, onChangePriority } = props;
  return (
    <div style={{ display: 'flex', flexDirection: 'column' }}>
      <InputLabel style={{ fontSize: 12 }}>
          Priority
      </InputLabel>
      <Select
        value={value}
        onChange={onChangePriority}
        autoWidth={false}
      >
        <MenuItem value="high">
           High
        </MenuItem>
        <MenuItem value="medium">
          Medium
        </MenuItem>
        <MenuItem value="low">
          Low
        </MenuItem>
      </Select>
    </div>
  );
}

SelectPriority.propTypes = {
  value: PropTypes.string.isRequired,
  onChangePriority: PropTypes.func.isRequired,
};
