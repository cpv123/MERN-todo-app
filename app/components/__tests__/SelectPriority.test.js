/* eslint-disable */
import React from 'react';
import { shallow } from 'enzyme';
import SelectPriority from '../SelectPriority.jsx';

describe('the SelectPriority component', () => {
  it('renders correctly', () => {
    const mockProps = { value: "High", onChangePriority: jest.fn() };
    const wrapper = shallow(<SelectPriority {...mockProps} />);
    expect(wrapper).toMatchSnapshot();
  });
});