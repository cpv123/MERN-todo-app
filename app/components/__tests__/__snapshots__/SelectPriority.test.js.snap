// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the SelectPriority component renders correctly 1`] = `
<div
  style={
    Object {
      "display": "flex",
      "flexDirection": "column",
    }
  }
>
  <WithStyles(InputLabel)
    style={
      Object {
        "fontSize": 12,
      }
    }
  >
    Priority
  </WithStyles(InputLabel)>
  <WithStyles(Select)
    autoWidth={false}
    onChange={[MockFunction]}
    value="High"
  >
    <WithStyles(MenuItem)
      key="High"
      value="High"
    >
      High
    </WithStyles(MenuItem)>
    <WithStyles(MenuItem)
      key="Medium"
      value="Medium"
    >
      Medium
    </WithStyles(MenuItem)>
    <WithStyles(MenuItem)
      key="Low"
      value="Low"
    >
      Low
    </WithStyles(MenuItem)>
  </WithStyles(Select)>
</div>
`;
